### DISCUSSION TITLE
Getting Entry with Asset but without S3 meta data

### DISCUSSION THREAD

**Discussion by barnabas-szekeres** [statamic/cms#6036] [Q&A]
Hi Folks,

Is there any solution to getting asset field url without meta data? I have an entry collection and these entires have thumbnails which are located on a S3 bucket. Unfortunately if query the `Entry` collection with `thumbnail` field the request goes very slow. :(

Example:

```
        $episodes = Entry::query()
            ->where('collection', 'episodes')
            ->where('published', true)
            ->get(['title', 'slug', 'thumbnail'])

        return $episodes;
```

This query result contain the whole asset's meta data, which is totally unnecessary for me. I only need the `thumbnail.url` but the query builder doesn't reflect to that.

```
[{
       "title": "Title",
       "slug": "slug",
        "thumbnail": {
            "id": "assets::XXXXX.png",
            "title": "XXXXX.png",
            "path": "XXXXX.png",
            "filename": "XXXXX",
            "basename": "XXXXX.png",
            "extension": "png",
            "is_asset": true,
            "is_audio": false,
            "is_previewable": false,
            "is_image": true,
            "is_video": false,
            "blueprint": {
                "title": "Assets",
                "handle": "assets"
            },
            "edit_url": "http://localhost/XXXXXX.png/edit",
            "container": {
                "id": "assets",
                "title": "Globális",
                "handle": "assets",
                "disk": "assets",
                "blueprint": {
                    "title": "Assets",
                    "handle": "assets"
                },
                "search_index": null,
                "api_url": null
            },
            "folder": "/",
            "url": "https://XXXXXX.png",
            "permalink": "https://XXXXXX.png",
            "api_url": "http://localhost/api/cms/assets/assets/XXXXXX.png",
            "size": "3.75 MB",
            "size_bytes": 3927781,
            "size_kilobytes": 3,
            "size_megabytes": 3.75,
            "size_gigabytes": 0,
            "size_b": 3927781,
            "size_kb": 3,
            "size_mb": 3.75,
            "size_gb": 0,
            "last_modified": "2022-05-07T08:54:23.000000Z",
            "last_modified_timestamp": 1651913663,
            "last_modified_instance": "2022-05-07T08:54:23.000000Z",
            "focus": "50-50-1",
            "has_focus": false,
            "focus_css": "50% 50%",
            "height": 1920,
            "width": 3840,
            "orientation": "landscape",
            "ratio": 2,
            "mime_type": "image/png",
            "duration": 0,
            "duration_seconds": 0,
            "duration_minutes": null,
            "duration_sec": 0,
            "duration_min": null,
            "playtime": "00:00"
        }
    }
}]
```

What I want to achieve:
```
[{
       "title": "Title",
       "slug": "slug",
       "thumbnail": {
           "url": "https://xxxx.png"
       }
}]
```

**Versions**
Statamic 3.3.11 Pro
Laravel 9
PHP 8.1

**Comment by joshuablum** [comment]
Hey @barnabas-szekeres! 👋

You could try using Laravel's `pluck()` method on the query result (which should be a Collection / EntryCollection). It also supports dot notation for retrieving nested values. So try something like `pluck('thumbnail.url')`. More in the Laravel docs [here](https://laravel.com/docs/9.x/collections#method-pluck).

Hope that helps! :)

**Reply by barnabas-szekeres** [reply]
Hey @joshuablum!

Thanks for your help! Sure, I'm familiar with the pluck method, however I need the whole collection with the thumbnail urls. Moreover pluck method will use the augmented version of the `thumbnail` field which has performance disadvantages :( 

I'm thinking of extending the Entry collection class some kind of method (like pluck) to iterate over my collection, because currently I have to do it in controllers every time, for example:

```
 // Iterate over the collection
 $episodes = $episodes->map(function($episode) {
     // Getting only the raw value and generate the url with Laravel's built in Storage facade
     $episode->set('thumbnail_url', Storage::disk('assets')->url($episode->get('thumbnail'));
     return $episode;
});
```

Do you have an idea how can I do this in more elegant way? In the Laravel world I would add an attribute to Entry model but the Entry class is not an eloquent model, so I cannot do that.

**Comment by jasonvarga** [comment]
Instead of immediately converting it all to an array, which will augment everything you've asked for, you could get the entries first and then just transform it into what you want. Similar to what you've already done. Something like this:

```php
Entry::query()
  ->get(['title', 'slug', 'assets_field'])
  ->map(function ($entry) {
    return [
      'title' => $entry->title,
      'slug' => $entry->slug,
      'thumbnail' => optional($entry->assets_field)->url,
    ];
  });
```

Also, do you have the stache watcher enabled in `config/statamic/stache.php`? If it's enabled, anything regarding assets will be slower, since it will constantly be checking the filesystem. As you're using S3, it would be particularly slow.

See how it feels if you disable it.
